tensorboard/webapp/metrics/store/metrics_reducers.ts

```diff
@@ -1081,14 +1096,32 @@ const reducer = createReducer(
         state.timeSeriesData,
         linkedTimeSelection
       );
+    const nextCardStateMap = {...state.cardStateMap};
+    if (cardId) {
+      if (!nextCardStateMap[cardId]) {
+        nextCardStateMap[cardId] = {};
+      }
+      nextCardStateMap[cardId].timeSelection = linkedTimeSelection;
+    }
 
     return {
       ...state,
       linkedTimeSelection,
       cardStepIndex: nextCardStepIndexMap,
+      cardStateMap: nextCardStateMap,
       rangeSelectionEnabled: nextRangeSelectionEnabled,
     };
   }),
+  on(actions.cardMinMaxChanged, (state, {cardId, minMax}) => {
+    const nextCardStateMap = {...state.cardStateMap};
+    nextCardStateMap[cardId] = {...nextCardStateMap[cardId]};
```

Comment:
This one might be ok but, I think, to be certain and to be consistent with other code, you could rewrite as:

```
nextCardStateMap[cardId] = {
 ...nextCardStateMap[cardId],
 userMinMax: minMax,
};
```